test/Mono.Linker.Tests.Cases/Reflection/MemberUsedViaReflection.cs

```diff
@@ -0,0 +1,487 @@
+﻿using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection;
+using Mono.Linker.Tests.Cases.Expectations.Assertions;
+using Mono.Linker.Tests.Cases.Expectations.Metadata;
+
+namespace Mono.Linker.Tests.Cases.Reflection
+{
+	[SetupCSharpCompilerToUse ("csc")]
+	public class MemberUsedViaReflection
+	{
+		public static void Main ()
+		{
+			// Normally calls to GetMember use regex to find values, we took a conservative approach
+			// and preserve not based on the string passed but on the binding flags requirements
+			TestWithName ();
+			TestWithRegex ();
+			TestWithBindingFlags ();
+			TestWithUnknownBindingFlags (BindingFlags.Public);
+			TestWithMemberTypes ();
+			TestNullType ();
+			TestDataFlowType ();
+			TestDataFlowWithAnnotation (typeof (MyType));
+			TestIfElse (true);
+		}
+
+		[RecognizedReflectionAccessPattern]
+		[Kept]
+		static void TestWithName ()
+		{
+			var members = typeof (SimpleType).GetMember ("memberKept");
+		}
+
+
+		[RecognizedReflectionAccessPattern]
+		[Kept]
+		static void TestWithRegex ()
+		{
+			var members = typeof (RegexType).GetMember ("*FoundViaRegex");
+		}
+
+		[RecognizedReflectionAccessPattern]
+		[Kept]
+		static void TestWithBindingFlags ()
+		{
+			var members = typeof (BindingFlagsType).GetMember ("*FoundViaRegex", BindingFlags.Public | BindingFlags.NonPublic);
+		}
+
+		[RecognizedReflectionAccessPattern]
+		[Kept]
+		static void TestWithUnknownBindingFlags (BindingFlags bindingFlags)
+		{
+			// Since the binding flags are not known linker should mark all members on the type
+			var members = typeof (UnknownBindingFlags).GetMember ("*FoundViaRegex", bindingFlags);
+		}
+
+		[RecognizedReflectionAccessPattern]
+		[Kept]
+		static void TestWithMemberTypes ()
+		{
+			// Here we took the same conservative approach, instead of understanding MemberTypes we only use
+			// the information in the binding flags requirements and keep all the MemberTypes
+			var members = typeof (TestMemberTypes).GetMember ("*FoundViaRegex", MemberTypes.Method, BindingFlags.Public);
```

Comment:
BTW: "*C" will not work if I read the code in CoreLib correctly - it can only handle prefixes, not suffixes.

Reply:
Yup, thanks for the heads up 
https://github.com/dotnet/runtime/blob/0026a202ab4087d237f6be8824dff322d3a7d6be/src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs#L2059